cpp/tests/test_poly_geometric_path.cpp

```diff
@@ -40,7 +38,7 @@ class ConstructPiecewisePoly : public testing::Test {
 
 TEST_F(ConstructPiecewisePoly, OutputValueHasCorrectDOF) {
   toppra::Vector pos = path.eval_single(0.5, 0);
-  ASSERT_THAT(pos.rows(), testing::Eq(2));
+  ASSERT_EQ(pos.rows(), 2);
```

Comment:
I am pretty sure they are equivalent. Why is this a bug?

Reply:
This was not the bug. I had a weird compilation error at some point, coming from somewhere in `gmock.h` and removing the `include <gmock.h>` fixed. I couldn't understand why, and as we don't really use it, this is satisfactory to me.
ASSERT_THAT coming from this file, I had to change it.